### PR TITLE
Fix unhashable type error message

### LINKED ISSUE #2800: Incompatibility for error message raised by `hash` built-in function with unhashable type
## Feature

In CPython side, `hash` raises `TypeError` with `unhashable type: '{}'`(e.g. `unhashable type: 'dict'`) message but RustPython's `hash` function raises with just `unhashable type`.

You can reference two code references of RustPython and CPython implementations:
https://github.com/RustPython/RustPython/blob/53b8067acd1af79c2081e9eb56586fa8955749fd/vm/src/slots.rs#L221-L228
https://github.com/python/cpython/blob/0fd27375cabd12e68a2f12cfeca11a2d5043429e/Objects/object.c#L776-L782

### CPython side

CPython 3.9.2

```
>>> hash(None)
285171546
>>> hash(0)
0
>>> hash(dict())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unhashable type: 'dict'
>>> hash(set())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unhashable type: 'set'
>>> 
```

### RustPython side

CPython 3.9.2

```
>>>>> hash(None)
140573235324752
>>>>> hash(0)
0
>>>>> hash(dict())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unhashable type
>>>>> hash(set())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unhashable type
```

## Python Documentation

https://docs.python.org/3.9/library/functions.html?highlight=hash#hash

### PR DESCRIPTION
**Pull request by moreal**
This pull request resolves #2800 by fixing the message format of the error raised by `hash` built-in function.

After this pull request, the message will have compatibility with CPython implementation and will have more rich information about error. 😎 

You can check the changes by executing below lines at [the current master branch commit](https://github.com/RustPython/RustPython/commit/53b8067acd1af79c2081e9eb56586fa8955749fd) and [this pull request HEAD commit](https://github.com/RustPython/RustPython/commits/fcfc15778d5937b7626ac39d2239a9eabfb85769).

```python
hash(dict())
hash(set())
```

### CODE REVIEW

File: vm/src/slots.rs [hunk 1]
```diff
@@ -223,7 +223,11 @@ where
     T: Unhashable,
 {
     fn hash(_zelf: &PyRef<Self>, vm: &VirtualMachine) -> PyResult<PyHash> {
-        Err(vm.new_type_error("unhashable type".to_owned()))
+        let msg = format!(
+            "unhashable type: '{}'",
+            _zelf.class().name
+        );
+        Err(vm.new_type_error(msg))
```
Comment (DimitrisJim) [inline vm/src/slots.rs:230]: nitpick: you can move format directly in the call here.

Comment (moreal) [reply vm/src/slots.rs:230]: Do you mean like:

```suggestion
        Err(vm.new_type_error(!format("unhashable type: '{}'", _zelf.class().name)))
```

Comment (DimitrisJim) [reply vm/src/slots.rs:230]: Yup!

Comment (moreal) [reply vm/src/slots.rs]: I applied it in a1d958b commit! 😉

**Review by DimitrisJim — APPROVED** [review]

**Comment by DimitrisJim** [comment]
Thanks!

**Pull request merged**